src/host/main.cpp

```diff
@@ -294,15 +298,44 @@ int main(int argc, char** argv)
         config.output_files.node_to_node_address_file);
     }
 
-    asynchost::RPCConnections rpc(
-      writer_factory, config.client_connection_timeout);
+    asynchost::ConnIDGenerator idGen;
+
+    asynchost::RPCConnections<asynchost::TCP> rpc(
+      writer_factory, idGen, config.client_connection_timeout);
     rpc.register_message_handlers(bp.get_dispatcher());
 
+    // This is a temporary solution to keep UDP RPC handlers in the same
+    // way as the TCP ones without having to parametrize per connection,
+    // which is not yet possible, due to UDP and TCP not being derived
+    // from the same abstract class.
+    asynchost::RPCConnections<asynchost::UDP> rpc_udp(
+      writer_factory, idGen, config.client_connection_timeout);
+    rpc_udp.register_quic_message_handlers(bp.get_dispatcher());
+
```

Comment:
Why do these share an `idGen`, rather than having their own? As long as we only ever use the ID within protocol-specific messages (ie, either `tls::` or `quic::`), it's fine if ID 5 is reused.

Reply:
I've factored that out after getting errors in the log saying "A connection with this ID already exists". I didn't did too deeply, though.

It may be conceptually fine for different protocols to have their own ID space, but the sessions' IDs are bundled somewhere in the middle, and I'm guessing on the enclave side that doesn't differentiate the protocol in `accept(id, ...)`.